src/App.tsx

```diff
@@ -0,0 +1,200 @@
+    const passwordLength: number = length[0];
+
+    let stringSet: string = "abcdefghijklmnopqrstuvwxyz";
+    if (capitalAllowed) stringSet += "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    if (numberAllowed) stringSet += "0123456789";
+    if (symbolAllowed) stringSet += `!"#$%&\:;<=>?@'({|}~)*+,-[\\]^_./`;
+
+    let tempPassword: string = "";
+    for (let i = 0; i < passwordLength; i++) {
+      const charIndex: number = Math.floor(Math.random() * stringSet.length);
+      tempPassword += stringSet[charIndex];
+    }
+    setPassword(tempPassword);
+    setCopied(false);
+  };
+
+  React.useEffect(() => {
+    generatePassword();
+  }, [capitalAllowed, numberAllowed, symbolAllowed, length]);
+
+  return (
+    <>
+      <Card
+        overrides={{
+          Root: {
+            style: {
+              maxWidth: "400px",
+              position: "absolute",
+              top: "20px",
+              right: "50%",
+              transform: "translate(50%, 0)",
+              width: "95%",
+              backgroundColor: "#121212",
+              border: "none",
+            },
+          },
+        }}
+      >
+        <StyledBody>
+          <Input
+            value={password}
+            ref={passwordRef}
+            onChange={(event) => setPassword(event.target.value)}
+            overrides={{
+              Root: {
+                style: {
+                  backgroundColor: "#121212",
+                },
+              },
+              Input: {
+                style: {
+                  color: "white",
+                  backgroundColor: "#121212",
+                  fontWeight: "bold",
+                  fontSize: "20px",
+                  letterSpacing: "1px",
+                },
+              },
+              InputContainer: {
+                style: {
+                  color: "white",
+                  backgroundColor: "#121212",
+                  fontWeight: "bold",
+                  fontSize: "20px",
+                  letterSpacing: "1px",
+                },
+              },
+              After: () => (
+                <Button shape={SHAPE.square} onClick={generatePassword}>
+                  <svg
+                    className={css({
+                      height: theme.sizing.scale800,
+                      width: theme.sizing.scale800,
+                    })}
+                    viewBox="0 0 24 24"
+                  >
+                    <path
+                      fill="#aaaaaa"
+                      d="M17.65,6.35C16.2,4.9 14.21,4 12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20C15.73,20 18.84,17.45 19.73,14H17.65C16.83,16.33 14.61,18 12,18A6,6 0 0,1 6,12A6,6 0 0,1 12,6C13.66,6 15.14,6.69 16.22,7.78L13,11H20V4L17.65,6.35Z"
+                    />
+                  </svg>
+                </Button>
+              ),
+            }}
+          />
+        </StyledBody>
+
+        <StyledAction>
+          <Button
+            onClick={() => {
+              navigator.clipboard.writeText(password);
```

Comment:
`writeText` returns a promise, you need to wait for the promise to resolve here.

```tsx
const copyToClipboard = async(): Promise<void> => {
  await navigator.clipboard.writeText(password);
  setCopied(true);
};
```